planner/core/planbuilder.go

```diff
@@ -692,8 +693,17 @@ func getPhysicalIDs(tblInfo *model.TableInfo, partitionNames []model.CIStr) ([]i
 }
 
 func (b *PlanBuilder) buildAnalyzeTable(as *ast.AnalyzeTableStmt) (Plan, error) {
+	checker := privilege.GetPrivilegeManager(b.ctx)
 	p := &Analyze{MaxNumBuckets: as.MaxNumBuckets}
 	for _, tbl := range as.TableNames {
+		// MySQL will analyze each table and stop on the first to error without privs
+		user := b.ctx.GetSessionVars().User
+		if checker != nil && user != nil &&
```

Comment:
How about does it in `buildAnalyze`? Or We also need to consider `buildAnalyzeIndex` and `buildAnalyzeAllIndex`.